### ISSUE TITLE
Should `to arrow`, `to parquet`, `<dataframe> | to csv` and `into sqlite` really be subcommands of `to` and `into` instead of `save`?

### ISSUE THREAD

**Issue by webbedspace** [nushell/nushell#6818]
### Question

The current subcommands of `to` look like this:
```
Subcommands:
  to arrow - Saves dataframe to arrow file
  to csv - Convert table into .csv text
  to csv - Saves dataframe to csv file
  to html - Convert table into simple HTML
  to json - Converts table data into JSON text.
  to md - Convert table into simple Markdown
  to nuon - Converts table data into Nuon (Nushell Object Notation) text.
  to parquet - Saves dataframe to parquet file
  to text - Converts data into simple text.
  to toml - Convert table into .toml text
  to tsv - Convert table into .tsv text
  to url - Convert table into url-encoded text
  to xml - Convert table into .xml text
  to yaml - Convert table into .yaml/.yml text
```
I find it very troubling that three of these do *not* output data, but instead take a filename and hit the disk. `to csv` is especially troubling - `<dataframe> | to csv <filename>` has an entirely different arity to `<table> | to csv` (that is, *based entirely on input type*).

Along with these, there is also `into sqlite <filename>`, which has the same issue.

I personally recommend that all file-making or file-changing operations be unambiguous - `save` and `touch` always hit the disk, always - and *not* be strewn within a basic data-conversion tool like `to`. My proposal is that they be added as subcommands of `save` - `save parquet <filename>`, `save csv <filename>`, `save arrow <filename>`, `save sqlite <filename>`.

Incidentally, these particular `to` commands seem to return a bizarre and rather useless datatype, `list<string>`, instead of an actual notification message.
```
〉ls | into df | to csv test.csv
╭───┬─────────────────────────────────────────╮
│ 0 │ saved "C:\\Users\\Leon\\TODO\\test.csv" │
╰───┴─────────────────────────────────────────╯
```
Note that `save` and `into sqlite` return `nothing` of the sort when they save a file.

**Comment by rgwood** [comment]
Thanks for the good issue - I agree that this is a problem, and I like the proposed solutions.

> `to csv` is especially troubling - `<dataframe> | to csv <filename>` has an entirely different arity to `<table> | to csv` (that is, *based entirely on input type*).

Agreed. I'm personally not convinced that we *should* be overloading commands based on input type, but if we are going to commit to that feature then commands with the same name should generally have the same signature.

**Comment by sholderbach** [comment]
I think this was solved by #13568 for the `to ...` command semantics and the conflict with the pure format converters has been resolved since we ditched the old command overloading.
